system/jlib/jfile.cpp

```diff
@@ -760,6 +760,69 @@ bool CFile::remove()
 #endif
 }
 
+static std::atomic<unsigned> renameRetries{(unsigned)-1}; // initially uninitialized, set on 1st use (in rename)
+
+void setRenameRetries(unsigned numRetries)
+{
+    renameRetries = numRetries;
+}
+
+static void doRename(const char *src, const char *dst, const char *errMsg)
+{
+    unsigned retriesLeft = renameRetries;
+    unsigned retry = 0;
+    while (true)
+    {
+        if ((unsigned)-1 == retriesLeft)
+        {
+            // NB: potentially could be >1 thread here, but that's ok.
+            try
+            {
+                retriesLeft = getGlobalConfigSP()->getPropInt("misc/@numRenameRetries");
+            }
+            catch (IException *e) // handle cases where config. not available
+            {
+                EXCLOG(e, "doRename");
+                e->Release();
+                retriesLeft = 0;
+            }
+            dbgassertex((unsigned)-1 != retriesLeft);
+            setRenameRetries(retriesLeft);
+        }
+        if (-1 != ::rename(src, dst))
+            return; // success
+
+        if (0 == retriesLeft)
+            throw makeErrnoExceptionV("%s(%s, %s)", errMsg, src, dst);
+
+#ifndef _WIN32 // NB: this is primary here to help track down the issue report in HPCC-28454
+        struct stat srcInfo;
```

Comment:
Could do with a comment describing what it is doing..

Reply:
will add a comment.